airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobPersistence.java

```diff
@@ -294,6 +294,29 @@ public void succeedAttempt(long jobId, int attemptNumber) throws IOException {
     });
   }
 
+  @Override
+  public void setAttemptTemporalWorkflowId(long jobId, int attemptNumber, String temporalWorkflowId) throws IOException {
+    database.query(ctx -> ctx.execute(
+        " UPDATE attempts SET temporal_workflow_id = ? WHERE job_id = ? AND attempt_number = ?",
+        temporalWorkflowId,
+        jobId,
+        attemptNumber));
```

Comment:
By the way, you can use the generate jOOQ code now, which should be easier to write than raw SQL query.

Reply:
haha I don't dare to do so since this PR already has a lot of moving pieces. I'll try it next time 😄